src/prototype_creep.js

```diff
@@ -167,9 +167,16 @@ Creep.prototype.stayInRoom = function() {
 };
 
 Creep.prototype.buildRoad = function() {
-  if (this.room.controller && this.room.controller.my && this.pos.lookFor(LOOK_TERRAIN)[0] !== 'swamp' &&
-    (this.room.controller.level < 4 || this.room.memory.misplacedSpawn)) {
-    return false;
+  if (this.room.controller && this.room.controller.my) {
+
+    if (this.pos.lookFor(LOOK_TERRAIN)[0] !== 'swamp' &&
+      (this.room.controller.level < 3 || this.room.memory.misplacedSpawn)) {
+      return false;
+    }
+    // TODO should be extracted to a method, stolen from role_harvester
+    if (!this.room.storage || this.room.storage.store.energy + 1000 < config.creep.energyFromStorageThreshold) {
```

Comment:
`this.room.controller.level < 3` always means that `!this.room.storage`, so we again wouldn't have roads on swamps in low RCL

Also I'm not sure this approach is good at all. I nearly always have low energy in storage, as upgrader spent it even faster then harvester fills extentions. We do not need a lot of energy for repairing roads, but if it'll decay, we'll fill storage even slower. So repair is mast have.
Do not build roads with low energy? May be... But I afraid that in low RCL rooms without other RCL 8 energy donor, compleat building roads will take much more time that it should.